Getting a js error when opening camera: can't access property "ideal", t.height is undefined

### What happened?

Hi, I'm using jitsi with firefox and a "Insta360 Link 2C: Insta360 Link" webcam. Used to work fine a few months back, possibly stopped working on upgrade from firefox `140.6.0esr-1~deb13u1` to `140.7.0esr-1~deb13u1`, at least that happened at around the same time,

I'm on a self-hosted jitsi installation, at meet.glasklar.is. I don't know which precise version if jitsi (sidenote: It would be really helpful with a button or similar on the meet landing page to display version numbers of both server and javascript components).

I'm not very familiar with javascript debugging, but I've tried to look around with the firefox debugging tools, after enabling "paus on exceptions". The exception is raised on a line `this._constraints.height = { ideal: constraints.height.ideal };` in the JitsiLocalTrack constructor in JitsiLocalTrack.ts.

I can see that the input constraints comes from the `tracksInfo` in `RTC.js:obtainAudioAndVideoPermissions`. If I look at the tracksInfo in the console, it's an array with a single entry
```
  0: Object { constraints: {…}, effects: (1) […], videoType: "camera", … }
    constraints: Object { frameRate: {…}, deviceId: {…} }
      deviceId: Object { exact: "aUewI4iG0ivY457MHrc9R7xZoAYprSX4ZexEpoWxiFk=" }
      frameRate: Object { max: 30, min: 15 }
    effects: Array [ {…} ]
      0: Object { _model: {…}, _options: {…}, _segmentationPixelCount: 36864, … }
      length: 1
    stream: MediaStream { id: "{db58788f-7074-4be6-919c-41e248eb57b2}", active: true, onaddtrack: null, … }
      active: true
      id: "{db58788f-7074-4be6-919c-41e248eb57b2}"
      onaddtrack: null
      onremovetrack: null
    track: MediaStreamTrack { kind: "video", id: "{72695dfa-4364-4b25-ade4-35784b11a104}", label: "Insta360 Link 2C: Insta360 Link", … }
      enabled: true
      id: "{72695dfa-4364-4b25-ade4-35784b11a104}"
      kind: "video"
      label: "Insta360 Link 2C: Insta360 Link"
      muted: false
      onended: function xo()
      onmute: null
      onunmute: null
      readyState: "live"
    videoType: "camera"
```

So the constraints here specify only framerate, no width and height. If I also look at the `this._constraints` member variable where the exception happens (just after it's been assigned from `track.getConstraints()`, in the console it looks like
```
Object { height: {…}, width: {…} }
  height: Object { ideal: NaN }
​  width: Object { ideal: NaN }
```

And that's as far as I get with debugging. It looks like this case isn't handled. Not sure if the NaN values should just be left as is here, or what kind of fallback would make sense. For lack of known version id, this is what lines 177-207 looks like in the JitsiLocalTrack.ts file shown by firefox' debugger:

```
        // Cache the constraints of the track in case of any this track
        // model needs to call getUserMedia again, such as when unmuting.
        this._constraints = track.getConstraints();

        if (mediaType === MediaType.VIDEO) {
            if (videoType === VideoType.CAMERA) {
                // Safari returns an empty constraints object, construct the constraints using getSettings.
                // Firefox in "fingerprint resistance mode" does a similar thing, except a `mediaSource` key is set.
                if (!this._constraints.height || !this._constraints.width) {
                    this._constraints = {
                        height: { ideal: this.getHeight() },
                        width: { ideal: this.getWidth() }
                    };
                }

                // If the constraints are still empty, fallback to the constraints used for initial gUM.
                if (!isValidNumber(this._constraints.height.ideal) && !isValidNumber(this._constraints.width.ideal)) {
                    this._constraints.height = { ideal: constraints.height.ideal };  // <---- this is where exception happens
                    this._constraints.width = { ideal: constraints.width.ideal };
                }
            }

            // Get the resolution from the track itself since we do not know what camera capability the browser has
            // picked for the given constraints, fallback to the constraints if MediaStreamTrack.getSettings() doesn't
            // return the height.
            this.resolution = this.getHeight();
            if (!isValidNumber(this.resolution) && this._constraints.height?.ideal) {
                this.resolution = this._constraints.height.ideal;
            }
            this.maxEnabledResolution = this.resolution;
        }
```


### Platform

- [ ] Chrome (or Chromium based)
- [x] Firefox
- [ ] Safari
- [ ] Other desktop browser
- [ ] Android browser
- [ ] iOS browser
- [ ] Electron app
- [ ] Android mobile app
- [ ] iOS mobile app
- [ ] Custom app using a mobile SDK

### Browser / app / sdk version

140.9.0esr (64-bit)

### Relevant log output

```shell
2026-03-26T12:20:04.669Z [INFO] [rtc:RTCUtils] <16/obtainAudioAndVideoPermissions/o<>:  Got media constraints:  {"video":{"frameRate":{"max":30,"min":15}},"audio":{"autoGainControl":true,"echoCancellation":true,"noiseSuppression":true}} Logger.js:158:27
2026-03-26T12:20:06.273Z [INFO] [app:redux] <persistState>:  redux state persisted. 75cf21309fca7fe98d0974491cf8ab21 -> 8bfce9eb2adbd7531b33424077340c2c Logger.js:158:27
2026-03-26T12:20:15.701Z [INFO] [rtc:RTCUtils] <16/_getUserMedia/</<>:  onUserMediaSuccess Logger.js:158:27
2026-03-26T12:20:15.708Z [ERROR] [app:tracks] <Ev/</<>:  Failed to create local tracks 
Array [ "audio", "video" ]
 TypeError: can't access property "ideal", t.height is undefined
    xo JitsiLocalTrack.ts:194
    Ia RTC.js:67
    Ia RTC.js:49
    obtainAudioAndVideoPermissions RTC.js:187
    promise callback*obtainAudioAndVideoPermissions RTC.js:187
    createLocalTracks JitsiMeetJS.ts:205
```

### Reproducibility

- [ ] The problem is reproducible on meet.jit.si

### More details?

_No response_

--- a/modules/RTC/RTCUtils.js
+++ b/modules/RTC/RTCUtils.js
@@ -110,7 +110,7 @@ function getConstraints(um = [], options = {}) {
     const constraints = cloneDeep(options.constraints || DEFAULT_CONSTRAINTS);
 
     if (um.indexOf('video') >= 0) {
-        if (!constraints.video) {
+        if (typeof constraints.video !== 'object' || constraints.video === null) {
             constraints.video = {};
         }
 
@@ -122,8 +122,14 @@ function getConstraints(um = [], options = {}) {
             constraints.video.width = { ideal: r.width };
         }
 
+        if (!constraints.video.height) {
+            constraints.video.height = cloneDeep(DEFAULT_CONSTRAINTS.video.height);
+        }
+        if (!constraints.video.width) {
+            constraints.video.width = cloneDeep(DEFAULT_CONSTRAINTS.video.width);
+        }
         if (!constraints.video.frameRate) {
-            constraints.video.frameRate = DEFAULT_CONSTRAINTS.video.frameRate;
+            constraints.video.frameRate = cloneDeep(DEFAULT_CONSTRAINTS.video.frameRate);
         }
 
         // Override the constraints on Safari because of the following webkit bug.

